Router fixes

kurdin opened the pull request:
**Objective**

This PR adds support for basename options for history and ensure that Link component creates correct href url with basename prefix based on history.createHref method in browser and using baseUrl prop on RouterContext for SSR.

Havunen commented:
Hi @kurdin thanks for PR. Can you please add tests to avoid regression in future

kurdin commented:
@Havunen sure I will do tests as well.

kurdin commented:
Hmm, I could not create tests before this https://github.com/infernojs/inferno/issues/669 gets fixed.  Have this error all over the place with url did not match any routes.

kurdin commented:
@Havunen I added one test to demonstrate how Link component works for basename. Also in router.spec.jsx tests I changed `createMemoryHistory` to `createBrowserHistory` because it is more relevant for using in browsers, you don't need `createMemoryHistory` at all.

Havunen commented:
Thank you. LGTM
